src/v/serde/time_point.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "serde.h"
+
+#include <seastar/core/lowres_clock.hh>
+
+namespace serde {
+
+// A serializable time_point type. Generally time points have a risk of
+// clock synchronization issues when deserialized on node with a clock drift.
+// Rather than letting users serialize arbitrary time_points, we only allow
+// serializing this special wrapped type making it easy for us to audit the
+// usage.
+// A time_point is serialized as the underlying duration epoch.
+template<class time_point_type = ss::lowres_clock::time_point>
```

Comment:
I guess I just don't really think this is a safe utility: 

[ss::lowres_clock](https://github.com/redpanda-data/seastar/blob/79a3038e90c45195a0446a067f516f7ff0ada0be/include/seastar/core/lowres_clock.hh#L47-L48
)
> This is a monotonic clock with a granularity of ~task_quota. Time points from this clock do not correspond to system time.



One could argue that it could be made safer by requiring the clock itself to have `to_time_t` / `from_time_t`.

Reply:
Made it safer:
```diff
--- a/src/v/serde/time_point.h
+++ b/src/v/serde/time_point.h
@@ -15,13 +15,20 @@
 
 namespace serde {
 
+template<typename T>
+concept system_clock_duration = requires {
+    // std::is_clock<T> - I don't think libc++ has this
+    T::clock::from_time_t;
+    T::clock::to_time_t;
+};
+
 // A serializable time_point type. Generally time points have a risk of
 // clock synchronization issues when deserialized on node with a clock drift.
 // Rather than letting users serialize arbitrary time_points, we only allow
 // serializing this special wrapped type making it easy for us to audit the
 // usage.
 // A time_point is serialized as the underlying duration epoch.
-template<class time_point_type = ss::lowres_clock::time_point>
+template<
+  system_clock_duration time_point_type = ss::lowres_system_clock::time_point>
 struct time_point : envelope<time_point<time_point_type>, version<0>> {
     using duration = typename time_point_type::duration;
     using clock = typename time_point_type::clock;
```

Reply:
Done, switched to lowres_system_clock as default.